public/app/features/alerting/unified/reducers/alertmanager/notificationPolicyRoutes.test.ts

```diff
@@ -0,0 +1,132 @@
+import { produce } from 'immer';
+
+import { AlertManagerCortexConfig, MatcherOperator, Route } from 'app/plugins/datasource/alertmanager/types';
+
+import { FormAmRoute } from '../../types/amroutes';
+import { addUniqueIdentifierToRoute } from '../../utils/amroutes';
+
+import { addRouteAction, deleteRouteAction, routesReducer, updateRouteAction } from './notificationPolicyRoutes';
+
+describe('routes', () => {
+  const defaultRoute: Route = {
+    receiver: 'ROOT',
+    group_by: ['grafana_folder'],
+    object_matchers: [],
+
+    routes: [
+      {
+        receiver: 'A',
+
+        object_matchers: [['team', MatcherOperator.equal, 'operations']],
+        routes: [
+          {
+            receiver: 'B1',
+            object_matchers: [['region', MatcherOperator.equal, 'europe']],
+          },
+          {
+            receiver: 'B2',
+            object_matchers: [['region', MatcherOperator.equal, 'nasa']],
+          },
+        ],
+      },
+      {
+        receiver: 'C',
+        object_matchers: [['foo', MatcherOperator.equal, 'bar']],
+      },
+    ],
+    group_wait: '10s',
+    group_interval: '1m',
+  };
+  const rootRouteWithIdentifiers = addUniqueIdentifierToRoute(defaultRoute);
+  const ROOT_identifier = rootRouteWithIdentifiers.id;
+  const A_identifier = rootRouteWithIdentifiers.routes![0].id;
+
+  const initialConfig: AlertManagerCortexConfig = {
+    alertmanager_config: {
+      time_intervals: [],
+      mute_time_intervals: [],
+      route: rootRouteWithIdentifiers,
```

Comment:
Should it already contain identifiers in this place?

Reply:
It should not! I'll update this :)